gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/modes/interferences/ApplyingAndSuppressingOrRemoveUnusedInterference.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.suppressibleerrorprone.modes.interferences;
+
+import com.google.common.collect.Sets;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.CommonOptions;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModeInterference;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModeInterferenceResult;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModeName;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+import one.util.streamex.EntryStream;
+
+/**
+ * Interference between Apply && (Suppress || RemoveUnused).
+ */
+public final class ApplyingAndSuppressingOrRemoveUnusedInterference implements ModeInterference {
+    private static final Set<ModeName> APPLY_AND_REMOVE_UNUSED_AND_SUPPRESS =
+            Set.of(ModeName.APPLY, ModeName.REMOVE_UNUSED, ModeName.SUPPRESS);
+    private static final Set<ModeName> APPLY_AND_REMOVE_UNUSED = Set.of(ModeName.APPLY, ModeName.REMOVE_UNUSED);
+    private static final Set<ModeName> APPLY_AND_SUPPRESS = Set.of(ModeName.APPLY, ModeName.SUPPRESS);
+
+    @Override
+    public ModeInterferenceResult interferesWith(Set<ModeName> modeNames) {
+        Optional<Set<ModeName>> maximalInterference = getMaximalInterference(modeNames);
+        return maximalInterference
+                .map(ModeInterferenceResult::interferenceBetween)
+                .orElseGet(ModeInterferenceResult::noInterference);
+    }
+
+    @Override
+    public CommonOptions interfere(Map<ModeName, CommonOptions> modeCommonOptions) {
+        Set<ModeName> maximalInterference =
+                getMaximalInterference(modeCommonOptions.keySet()).get();
+        CommonOptions apply = modeCommonOptions.get(ModeName.APPLY);
+        CommonOptions naivelyCombined = EntryStream.of(modeCommonOptions)
+                .filterKeys(maximalInterference::contains)
+                .values()
+                .reduce(CommonOptions.empty(), CommonOptions::naivelyCombinedWith);
```

Comment:
I think this is only going to give you the mode names that you returned in `interferesWith`, so you can just combine them all then add the extra flag - no need to calculate the maximal interference again?